src/kudu/rpc/reactor.cc

```diff
@@ -342,6 +342,20 @@ void ReactorThread::RegisterConnection(scoped_refptr<Connection> conn) {
   server_conns_.emplace_back(std::move(conn));
 }
 
+void ReactorThread::ResetAllConnections() {
+    DCHECK(IsCurrentThread());
+    for (const scoped_refptr<Connection> &conn : server_conns_) {
+        conn->Shutdown(
+                Status::Aborted("Shutting down server connection by request"));
```

Comment:
This is slightly less graceful today since we just force the conn to shutdown. The leader will reestablish the connection soon enough though. (verified this).

A better way to do this is to wait on an inbound transfer to finish (but response can still be discarded so leader is going to resend the batch anyway). To wait for even the response to finish would be a lot more complicated since the outbound stream is generally treated as another stream.

I'm thinking for a manual conn reset we can just afford to shutdown the conn as there isn't actually a "good" time to do it that's a lot better than this.